ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/commands/CommandCompletionServiceIT.scala

```diff
@@ -154,6 +162,64 @@ class CommandCompletionServiceIT
             commandIds3 should not contain (commandIds1(1))
           }
       }
+
+      "implicitly tail the stream if no offset is passed" in allFixtures { ctx =>
+        // A command to create a Test.Dummy contract (see //ledger/sandbox/src/main/resources/damls/Test.daml)
+        val createDummyCommand = ProtoCommand(
+          ProtoCommand.Command.Create(
+            ProtoCreateCommand(
+              templateId = Some(templateIds.dummy),
+              createArguments = Some(
+                Record(
+                  fields = Seq(
+                    RecordField("operator", Some(Value(Value.Sum.Party(party))))
+                  )
+                ))
+            )))
+
+        // Issue the `createDummyCommand` to the ledger using the managed command client
+        def createDummy(id: String) =
+          for {
+            commandClient <- ctx.commandClient(ctx.ledgerId)
+            request = ctx.command(id, Seq(createDummyCommand))
+            _ <- commandClient.submitSingleCommand(request)
+          } yield ()
+
+        // Issues several `createDummyCommand`s
+        def createDummies(ids: List[String]) =
+          Future.sequence(ids.map(createDummy)).map(_ => ())
+
+        // Starts "tailing" the completions service and then issues one more command
+        def tailCompletionsAndConcurrentlyCreateDummyWithCommandId(id: String) = {
+          val completions = completionSource(
+            ctx.commandCompletionService,
+            ctx.ledgerId,
+            applicationId,
+            Seq(party),
+            offset = None).runWith(Sink.queue())
```

Comment:
to get only first element you should rather use `Sink.head` instead of a queue

Reply:
Yeah, I went through various iterations and in the end I forgot to restore `Sink.head`. Good catch, thanks! :slightly_smiling_face:

Reply:
I'm not entirely sure why but the tests seem to fail using `Sink.head`. I'll leave this as is it for now. Is it worth investigating this?

Reply:
I think it is. Using a queue here is misleading and an overkill. Can it be because of pure `Checkpoint`s? This actually reminds me that you need to filter the results stream not to get pure `Checkpoint` responses. There is a timing flaw here which can lead to flaky tests.